### PR TITLE
sync: remove crash! macro

### PR DESCRIPTION
**Pull request by nandhinianandj**
remove crash! macro.. beginner issue addresses https://github.com/uutils/coreutils/issues/5487 .

### CODE REVIEW

File: src/uu/sync/src/sync.rs [hunk 1]
```diff
@@ -88,12 +87,14 @@ mod platform {
             match OpenOptions::new().write(true).open(sliced_name) {
                 Ok(file) => {
                     if FlushFileBuffers(file.as_raw_handle() as HANDLE) == 0 {
-                        crash!(GetLastError() as i32, "failed to flush file buffer");
+                        USimpleError::new(1, format_error_message(GetLastError() as i32,
```
Comment (sylvestre) [inline src/uu/sync/src/sync.rs:90]: please run rustfmt, thanks

Comment (nandhinianandj) [reply src/uu/sync/src/sync.rs]: @sylvestre : Done with new commits. now all tests pass locally(on ubuntu linux) but fail on windows and android. Please help resolve those.

Comment (nandhinianandj) [reply src/uu/sync/src/sync.rs]: found it i need to explicity define a fomat_error_message function... as it is in src/uu/du/src/du.rs.. though not sure why it succeeded on the other linux instances.. adding that function. now.

**Review by tertsdiepraam — CHANGES_REQUESTED** [review]
The `windows` CI is failing unfortunately. I think, the `format_error_message` and `USimpleError` items need to be `use`d in the `platform` module. Otherwise, I think this is the right approach for now. Ultimately, we might decide to implement `UError` on `ParseSizeError`, but this is fine for now.

**Comment by sylvestre** [comment]
@nandhinianandj ping?

**Pull request closed without merging**